### DISCUSSION TITLE
some questions about hooks

### DISCUSSION THREAD

**Discussion by emrahcom** [ory/kratos#1953] [Q&A] [answered]
I read [Hooks](https://www.ory.sh/kratos/docs/self-service/hooks/), [Configuration](https://www.ory.sh/kratos/docs/reference/configuration/) and [Integration using Web-Hooks](https://www.ory.sh/kratos/docs/guides/integration-with-other-systems-using-web-hooks) topics but some points are still not clear for me.

- [Hooks](https://www.ory.sh/kratos/docs/self-service/hooks/) talks about two options for `After registration`.
  `Before persisting` and `After persisting`... I don't understand how I can set this.

- Is it possible to run a shell script directly as hook on the `kratos` host?

- Is there a list of hook types?
  `web_hook`, `session`, `revoke_active_sessions`, `require_verified_address`...
  what else..?

thanks

**Comment by emrahcom** [comment, marked as answer]
The topic is clear for me now. I marked as answered

**Reply by vsolomasov** [reply]
Hello! Is it possible to set webhook `Before persisting` for `After registration`?

**Reply by emrahcom** [reply]
AFAIK all options are in [the configuration page](https://www.ory.sh/kratos/docs/reference/configuration/)
You can add a before hook for the registration in your `kratos.yml`. If it fails, the registration will be aborted

```json
registration:
  before:
    hooks:
      - hook: web_hook
        config:
          url: http://address/hook
          method: POST
          body: /your/jsonnet/folder/filename.jsonnet
```

**Reply by charlie-whight** [reply]
Maybe I am missing it, but I am still unclear as to whether you can or cannot set "after persisting" or before persisting" as separate options in the "after registration" hooks

The docs seem to suggest you can but does not state how?

**Reply by alexeevaDariaIDT** [reply]
> Maybe I am missing it, but I am still unclear as to whether you can or cannot set "after persisting" or before persisting" as separate options in the "after registration" hooks
> 
> The docs seem to suggest you can but does not state how?

Same question here - as of now I wasn't able to set up a hook specifically "before persisting" identity in the database. 
Maybe if smn did they can provide me with an answer (example)

**Reply by aeneasr** [reply]
Hello, this is unfortunately not yet possible. There is an open PR to make it possible to update identity's traits as part of the web hook request, but it is not yet merged!
